### PR TITLE
Fix documentation problems

### PR DESCRIPTION
**Pull request by paulromano**
Most are pretty obvious. The list of reaction types unfortunately has to be a single line to render correctly in the generated documentation. I'm not sure if there was a reason you only marked the MGXS class under `:members:`; I've removed it as the subclasses should appear in documentation as well.

### CODE REVIEW

**Comment by wbinventor** [comment]
Thanks, I'll look at this momentarily. I picked out MGXS because Sphinx
appears to only know how to list classes in alphabetical order. Hence, MGXS
shows up in the middle of the docs, with a bunch of its subclasses
scattered about. I felt this like this ordering rendered the API docs
effectively useless. Of course, I don't think my solution was a great one -
all of the subclasses should be there as well - but I didn't dig in yet to
figure out a better approach.

**Comment by paulromano** [comment]
Ah, got it. Easy approach is to just use autoclass for each one (just committed a change that does exactly that) rather than automodule.

**Comment by wbinventor** [comment]
Awesome, I like the autoclass option. Sorry that you had to do my Sphinx homework for me.

File: docs/source/pythonapi/mgxs.rst [hunk 1]
```diff
@@ -4,5 +4,38 @@
 Multi-Group Cross Sections
 ==========================
 
-.. automodule:: openmc.mgxs.mgxs
-    :members: MGXS
+.. autoclass:: openmc.mgxs.mgxs.MGXS
```
Comment (wbinventor) [inline docs/source/pythonapi/mgxs.rst]: Along the same lines, do you know if it is (easily) possible to list a "table of contents" at the top of the page with links to each of the classes below?

**Comment by paulromano** [comment]
No problem- only took me a few minutes!

File: docs/source/pythonapi/mgxs.rst (hunk 1 above)
Comment (paulromano) [inline docs/source/pythonapi/mgxs.rst]: I didn't until now -- `sphinx.ext.autosummary` provides a nice way to do it.

**Comment by wbinventor** [comment]
The summary looks great, we might consider using it in some of the other pieces of our documentation, esp. if it will summarize a list of the routines available (*e.g.*, in the `Tally` class).

**Comment by paulromano** [comment]
Agreed- I had the same thought. It really would be useful in other places.

**Comment by wbinventor** [comment]
Yeah. On a related not to this, as I may have mentioned it before, I think the high bar for documentation is set by the [scikit-learn](http://scikit-learn.org/) project. One of the reasons for their insane success in the last 4-5 years is their insanely great API documentation. Since it's all open source, we leverage some of their best practices as we (slowly) move forward to improve our own docs.

**Pull request merged**